experimental/sdk/webpubsub-socketio-extension/src/EIO/components/web-pubsub-transport.ts

```diff
@@ -103,5 +131,86 @@ export class WebPubSubTransport extends Transport {
         throw error;
       }
     }
+    debug(`webPubSubSend, finish`);
+  }
+
+  private _isMessageWithBinary(packet: EioPacket): boolean {
+    return packet.data instanceof ArrayBuffer || ArrayBuffer.isView(packet.data);
+  }
+
+  private _isTypeWithBinary(packet: SioPacket): boolean {
+    return packet.type === SioPacketType.BINARY_EVENT || packet.type === SioPacketType.BINARY_ACK;
+  }
+
+  /**
+   * For SIO mode:
+   *  returns the largest `n` satisfying `packets[0 .. n - 1]` doesn't contain any incomplete binary-related EIO packet sequence.
+   *  A complete binary-related EIO packet sequence = [BINARY_EVENT/BINARY_ACK SIO payload, ...(all its binary attachments)]
+   * For EIO mode:
+   *  The number of binary attachments cannot be inferred. This method returns `packets.length` directly.
+   * @param packets - EIO packets
+   * @returns the number of packets that can be sent in the next REST API call
+   */
+  private _getPacketNumberForNextSend(packets: EioPacket[]): number {
+    if (!this._sioMode) return packets.length;
+
+    /**
+     * Binary Packet = an EIO packet whose data is BINARY_EVENT or BINARY_ACK SIO packet.
+     * Binary Attachment Packet = an EIO packet whose data is binary attachment of a binary packet.
+     */
+    let attachmentCount = 0, // the number of binary attachment packets found for the latest binary packet.
+      expectedAttachments = 0, // the expected number of binary attachment packets for the latest binary packet.
+      lastBinaryMessagePacketIdx = -1, // the index of the latest binary packet.
+      lastSentPacketIdx = -1, // the largest index of EIO packet that can be sent in the next REST API call
+      shouldBeAttachment = false; // whether the current packet should be a binary attachment packet
+
+    for (let i = 0; i < this._buffer.length; i++) {
+      const eioPacket = this._buffer[i];
+
+      // Condition 0: a pure EIO packet without data related to SIO packet.
+      if (eioPacket.type !== "message") {
+        lastSentPacketIdx++;
+        continue;
+      }
+
+      // Condition 1: A binary attachment packet
+      if (this._isMessageWithBinary(eioPacket)) {
+        if (!shouldBeAttachment) throw new Error("Expect a packet whose data is binary attachment but not found.");
```

Comment:
is the error correctly handled outside?

Reply:
added error handling outside. If error, send all packets to the service.